### DISCUSSION TITLE
[HTTP3] Add QUIC support

### DISCUSSION THREAD

**Discussion by iammarco11** [abhinavsingh/proxy.py#807] [Ideas]
I would love to see HTTP/3 support for the traffic monitoring by proxy :)

**Comment by abhinavsingh** [comment]
@iammarco11 Thanks for raising the request.  Last time I checked _(about an yr back)_ `http/3` spec, it wasn't clear about how proxies must handle `http/3` data frames.  However, this seems to have changed now.  Found following reference in `http/3` spec here https://tools.ietf.org/id/draft-ietf-quic-http-23.html#the-connect-method

<img width="1082" alt="Screen Shot 2020-10-02 at 12 07 37 PM" src="https://user-images.githubusercontent.com/126065/94894989-15b2fa80-04a8-11eb-9efb-99cedbd9c891.png">

Let's use this issue as a tracker to include `http/3` support in future milestones :)

**Comment by iammarco11** [comment]
Okay so are you starting the work?

**Comment by abhinavsingh** [comment]
@iammarco11 Hi Akshay, thanks for interest in this feature.  I won't have time on my hands to dive into a new feature anytime soon, specially something as exhaustive as `http/3`.  Currently, `h3` is used by around 7% sites in the world `(per Oct'2020 survey report)`.  Also, when using a proxy server, browsers _(and applications)_ will not use `http/3` protocol and simply fallback to `http/2` or `http/1.1`.  I am hoping some good folks will contribute `h3` features into the library :). If not, I might jump into it someday.

<img width="336" alt="Screen Shot 2020-10-29 at 10 15 04 PM" src="https://user-images.githubusercontent.com/126065/97605533-be676200-1a34-11eb-86d6-d757968e0a8f.png">

I would also like to understand if this feature is blocking you from using `proxy.py` in any way.  Thank you!!!

**Comment by iammarco11** [comment]
Well, I would like to know if you know some browsers which would support the HTTP/3 proxy. I am actually doing a research with different protocols, thought of having something similar to mitmproxy to visualise the traffic, wireshark works but looking for alternative proxy methods :)

**Comment by abhinavsingh** [comment]
> Well, I would like to know if you know some browsers which would support the HTTP/3 proxy. I am actually doing a research with different protocols, thought of having something similar to mitmproxy to visualise the traffic, wireshark works but looking for alternative proxy methods :)

I haven't experienced any issues without `h3` support in proxy.py yet.  My local browsing experience remains the same.
